switchtec.c

```diff
@@ -208,6 +208,17 @@ static void mrpc_complete_cmd(struct switchtec_dev *stdev)
 	stuser_set_state(stuser, MRPC_DONE);
 	stuser->return_code = 0;
 
+	if (stuser->status == SWITCHTEC_MRPC_STATUS_ERROR) {
```

Comment:
Why not just move the STATUS_DONE check down instead of duplicating the code that copies the return code?